### PR TITLE
gt_concordance

### PR DESCRIPTION
**Pull request by kate-v-stepanova**

### CODE REVIEW

File: scripts/gt_concordance.py [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+from ngi_pipeline.database.classes import CharonSession, CharonError
+from ngi_pipeline.log.loggers import minimal_logger
+from ngi_pipeline.utils.classes import with_ngi_config
+
+log = minimal_logger(os.path.basename(__file__))
+
+@click.group()
+@with_ngi_config
+@click.pass_context
+@click.option('--config', '-c', 'custom_config', help='Path to a config file', type=click.Path())
```
Comment (vezzi) [inline scripts/gt_concordance.py]: this is the same config file used when running ngi_pipeline, so better you default to that value using the $NGI_CONFIG  variable

Comment (vezzi) [inline scripts/gt_concordance.py]: also this option should be mandatory

Comment (kate-v-stepanova) [inline scripts/gt_concordance.py]: we use `@with_ngi_config`, which is a function that takes a config from ngi_pipeline. But  `--config` is an additional option, if we want to specify another config file.

File: scripts/gt_concordance.py [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+                gatk_var_file=config.get('GATK_VAR_FILE'))
+        full_command = 'java -Xmx6g -jar {} {}'.format(config.get('GATK_PATH'), options)
+        try:
+            subprocess.call(full_command.split())
+        except:
+            pass
+        else:
+            return output_file
+
+def update_charon(sample_id, status, concordance=None):
```
Comment (vezzi) [inline scripts/gt_concordance.py]: suggestion: call this function `update__genotype_status_charon ` or something similar,`update_charon ` is a bit too general and confusing

File: scripts/gt_concordance.py [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+                bamfile=bamfile,
+                sample=output_file,
+                interval_file=config.get('INTERVAL_FILE'),
+                gatk_ref_file=config.get('GATK_REF_FILE'),
+                gatk_var_file=config.get('GATK_VAR_FILE'))
+        full_command = 'java -Xmx6g -jar {} {}'.format(config.get('GATK_PATH'), options)
+        try:
+            subprocess.call(full_command.split())
+        except:
+            pass
```
Comment (vezzi) [inline scripts/gt_concordance.py]: I think you are being a bit inconsistent with return types, if the bam does not exists you return None, is you rise an exception here you do pass, therefore you go at the end of the function and you should by default retunr None, I prefer when the return objects are explicitly returned

File: scripts/gt_concordance.py [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+    result += '\nLost snps: {}\n'.format(len(lost))
+    result += 'Total number of matches: {} / {} / {}\n'.format(len(matches), len(vcf_data)-len(lost), len(vcf_data))
+    result += 'Percent matches {}%\n'.format(percent_matches)
+
+
+    # path should exist (if we came to this point), but checking anyway
+    output_path = os.path.join(config.get('ANALYSIS_PATH'), project, 'piper_ngi/03_genotype_concordance')
+    if os.path.exists(output_path):
+        # create .conc file
+        with open(os.path.join(output_path, '{}.conc'.format(sample)), 'w+') as conc_file:
```
Comment (vezzi) [inline scripts/gt_concordance.py]: we want to append results? maybe but not sure about the use-case

File: scripts/gt_concordance.py [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+            conc_file.write(result)
+
+    if len(matches) + len(mismatches) + len(lost) != len(vcf_data):
+        log.warning('CHECK RESULTS!! Numbers are incoherent. Total number of positions: {}, matches: {}, mismatches: {}, lost: {}'.format(len(vcf_data), len(matches), len(mismatches), len(lost)))
+
+    if len(lost) >= 30:
+        log.warning('Too few positions in VCF file!! Failed to caclulate concordance')
```
Comment (vezzi) [inline scripts/gt_concordance.py]: there is the risk to have a percent_matches of 100% but have lost > 30, in this case we have only the log information to find out this.... not sure there is an easy solution for this

Comment (vezzi) [inline scripts/gt_concordance.py]: maybe you should do the same thing you do when you do not have enough position, you foce to 0 the concordance

**Comment by vezzi** [comment]
Awesome @kate-v-stepanova , only minor comments see if and how you want to implement and then I can merge!!!!

File: scripts/gt_concordance.py (hunk 4 above)
Comment (kate-v-stepanova) [inline scripts/gt_concordance.py]: No, this one will overwrite the content, or create a new file if it doesn't exist

**Comment by vezzi** [comment]
Perfect, update the version please and then I can merge. @sylvinite this will be add to the next deployment, no need to test as it is detached from the production part

**Comment by vezzi** [comment]
GREAT WORK @kate-v-stepanova

**Comment by vezzi** [comment]
@sylvinite hav a look to the readme, new fields are needed in the stockholm specific config file for ngi_pipeline

**Pull request merged**